python/python/bystro/covariance/_covariance_np.py

```diff
@@ -0,0 +1,90 @@
+"""
+This implements two objects, the empirical covariance estimator used as a 
+baseline comparison method. It also implements BayesianCovariance, which
+implements MAP estimation using several common priors.
+
+Objects
+-------
+EmpiricalCovariance(BaseCovariance)
+    This object just fits the covariance matrix as the standard sample
+    covariance matrix
+
+BayesianCovariance(BaseCovariance):
+    This object fits the covariance matrix as the MAP estimator using
+    user-defined priors.
+"""
+import numpy as np
+from bystro.covariance._base_covariance import BaseCovariance
+
+
+class EmpiricalCovariance(BaseCovariance):
+    def __init__(self):
+        """
+        This object just fits the covariance matrix as the standard sample
+        covariance matrix
+        """
+        super().__init__()
+
+    def fit(self, X):
```

Comment:
for all public methods where we accept an `X`, what would you think about checking that X is centered and raise an informative ValueError if not?

Reply:
What should that check be? standardScalar for example only gets it to 1e-16 within 0 usually, and we dont' assume normalization. If the mean is .0001 for a single covariate does it fail?

Reply:
Thought I responded but didn't save for some reason. What check do you have in mind, since sklearn's standardscaler often gives values like 1e-16 for covariates. Is mean of .0001 too much if it's in a single covariate? What if the variance in that covariate is .001 instead of approximately 1? In my strong opinion this is one of those things where we have to trust the user to properly use the product.

Reply:
I think whether normal machine precision is acceptable for this application is an implementation detail I'd defer to your judgment on, my concern is more about gross departures from centeredness.  A reasonable compromise might be checking that `X` is `np.close` to its StandardScaled version.

Want to discuss this stuff on a call tomorrow?  I'm worried I just don't have enough context yet to understand the approach.

Reply:
Sure. But here's a straightforward application where np.close will fail, if you standardize a dataset before dividing it into a training and test set.